torch/distributed/_tensor/ops/embedding_ops.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+# implement matrix related ops for distributed tensor
+
+import torch
+
+from torch.distributed._tensor.api import DTensorSpec, Replicate, Shard
+from torch.distributed._tensor.op_schema import OpSchema, OutputSharding
+from torch.distributed._tensor.ops.utils import register_prop_rule
+
+aten = torch.ops.aten
+
+
+@register_prop_rule(aten.embedding.default)
+def embedding_rules(op_schema: OpSchema) -> OutputSharding:
+    weight_spec, inp_spec = op_schema.args_spec
+    if any(placement.is_shard(0) for placement in weight_spec.placements):
+        raise NotImplementedError(
+            "DTensor does not support row-wise sharded embedding operation yet!"
+        )
+
+    if all(placement.is_replicate() for placement in inp_spec.placements):
+        return OutputSharding(
+            output_spec=DTensorSpec(
+                mesh=inp_spec.mesh,
+                placements=[Replicate()] * (len(inp_spec.placements) - 1) + [Shard(-1)],  # type: ignore[list-item]
+                tensor_meta=inp_spec.tensor_meta,
```

Comment:
nit: for `output_sharding`, we don't need to set up the tensor meta field (can just leave it as None), as it would be overwritten by the sharding propagator itself.